P0009/P0009.md

```diff
@@ -1011,6 +1059,70 @@ for supporting heterogeneous memory.
 For a discussion of the idea of accessors and several examples,
 please see (Keryell and Falcou 2016).
 
+### Why do we distinguish between accessors and pointers?
+
+Our design orthogonalizes storage type or allocation handle (the pointer)
+from access properties (the accessor).
+An `mdspan` stores a `pointer`,
+but uses its accessor for element access and for making offset views.
+Just as `atomic_ref` and ordinary pointer dereferencing
+can access the same memory in two different ways,
+two different `mdspan` objects might access the same memory in two different ways.
+
+This explains why the accessor policy uses `a.access(p, i)`
+instead of `p[i]` to access the pointer `p`.
+For example, the `access` method can be decorated with nonstandard attributes
+for execution only on accelerators (e.g., GPUs).
+This would prevent "dereferencing" a device pointer on host, for example.
+The `pointer` type might just be `element_type*` in that case,
+so `p[i]` may not have the right attributes to make that distinction.
+
+#### Why store the pointer separately from the accessor?
+
+`mdspan` stores its pointer separately from the accessor.
+This has the following advantages:
+
+1. one can construct two different `mdspan`s with the same pointer,
+    even if they have different accessor types
+    (with the same `pointer` type);
+
+2. the accessor need not be stateful;
+
+3. `mdspan` is default constructible, even if the accessor is not; and
```

Comment:
?? It is not since the accessor member needs to be default constructible for mdspan to be.